### ISSUE TITLE
Unexpected error "no-unused-vars"

### ISSUE THREAD

**Issue by crossrw** [eslint/eslint#11541]
**My environment**

* **ESLint Version: 5.15.3**
* **Node Version: 10.15.3**
* **npm Version: 6.4.1**

**What parser (default, Babel-ESLint, etc.) are you using?**

**Full configuration:**

<details>
<summary>Configuration</summary>

```js
{
    "env": {
        "commonjs": true,
        "es6": true,
        "node": true
    },
    "extends": "eslint:recommended",
    "globals": {
        "Atomics": "readonly",
        "SharedArrayBuffer": "readonly"
    },
    "parserOptions": {
        "ecmaVersion": 2018
    },
    "rules": {
    }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
let buf = Buffer.allocUnsafe(13);
let offset = buf.writeUInt32LE(1, 0);
offset = buf.writeUInt16LE(2, offset);
```

```bash
eslint test.js
```

**What did you expect to happen?**

I expected no problems, since the variable "offset" is used in the method "writeUInt16LE".

**What actually happened? Please include the actual, raw output from ESLint.**

2:5  error  'offset' is assigned a value but never used  no-unused-vars
✖ 1 problem (1 error, 0 warnings)
![issue](https://user-images.githubusercontent.com/6500284/54844104-1aa9c200-4ce7-11e9-8af2-604ce5ed33ff.PNG)

**Comment by platinumazure** [comment]
Hi @crossrw, thanks for the issue.

I think this is, more or less, working as designed. The `offset` variable is assigned to on the last line but not used. Your code would be the same as this:

```js
let buf = Buffer.allocUnsafe(13);
let offset = buf.writeUInt32LE(1, 0);
buf.writeUInt16LE(2, offset);
```

I'm fairly sure this would be enough to ensure ESLint considers the variable as used.

The report location might be a bit confusing since it is reporting the declaration rather than the last assignment. Not 100% sure if that part might be a bug or not.

**Comment by not-an-aardvark** [comment]
I think the report location is currently as intended (see https://github.com/eslint/eslint/issues/7282).

That said, looking at this again it does seem like reporting the declaration site is fairly confusing in this case. It seems like it would be better to report the area around `offset = ` on the last line, with a message like `'offset' is never used after this assignment`. (I haven't thought through whether this solution would generalize well in more complex cases.)

**Comment by crossrw** [comment]
Yes, you are right, I was confused that the error message was not in the last line. It would be, in my opinion, more correct. Thank you!
